dids/src/main/kotlin/web5/sdk/dids/did/BearerDid.kt

```diff
@@ -0,0 +1,143 @@
+
+    val signer: DidSigner = { payload ->
+      keyManager.sign(keyAlias, payload)
+    }
+
+    return Pair(signer, verificationMethod)
+  }
+
+  /**
+   * Converts a `BearerDid` object to a portable format containing the URI and verification methods
+   * associated with the DID.
+   *
+   * This method is useful when you need to represent the key material and metadata associated with
+   * a DID in format that can be used independently of the specific DID method implementation. It
+   * extracts both public and private keys from the DID's key manager and organizes them into a
+   * `PortableDid` structure.
+   *
+   * @returns A `PortableDid` containing the URI, DID document, metadata, and optionally private
+   *          keys associated with the `BearerDid`.
+   */
+  public fun export(): PortableDid {
+
+    val keyExporter = keyManager as? KeyExporter
+    val privateKeys = mutableListOf<Jwk>()
+
+    document.verificationMethod?.forEach { vm ->
+      val keyAliasResult = runCatching { vm.publicKeyJwk?.computeThumbprint() }
+      if (keyAliasResult.isSuccess) {
+        val keyAlias = keyAliasResult.getOrNull()
+        keyExporter?.exportKey(keyAlias!!.toString())?.let { key ->
+          privateKeys.add(key)
+        }
+      }
+    }
+
+    return PortableDid(
+      uri = this.did.uri,
+      document = this.document,
+      privateKeys = privateKeys,
+      metadata = mapOf()
+    )
+  }
+
+  public companion object {
+
+
+    /**
+     * Instantiates a [BearerDid] object from a given [PortableDid].
+     *
+     * This method allows for the creation of a `BearerDid` object using a previously created DID's
+     * key material, DID document, and metadata.
+     *
+     * @param portableDid - The PortableDid object to import.
+     * @param keyManager - Optionally specify an external Key Management System (KMS) used to
+     *                     generate keys and sign data. If not given, a new
+     *                     [LocalKeyManager] instance will be created and used.
+     * @returns [BearerDid] object representing the DID formed from the
+     *          provided PortableDid.
+     */
+    @JvmOverloads
+    public fun import(
+      portableDid: PortableDid,
+      keyManager: KeyManager = InMemoryKeyManager()
+    ): BearerDid {
+      check(portableDid.document.verificationMethod?.size != 0) {
+        "PortableDID must contain at least one verification method"
+      }
+
+      val allVerificationMethodsHavePublicKey =
+        portableDid.document.verificationMethod
+          ?.all { vm -> vm.publicKeyJwk != null }
+          ?: false
+      check(allVerificationMethodsHavePublicKey) {
+        "Each VerificationMethod must contain a public key in Jwk format."
+      }
+
+      val did = Did.parse(portableDid.uri)
+
+      for (key in portableDid.privateKeys) {
+        val keyImporter = keyManager as? KeyImporter
+        keyImporter!!.importKey(key)
+      }
```

Comment:
`if (keyManager is KeyImporter)`

Reply:
i put a check at the top `check(keyManager is KeyImporter) { "error" }`